src/api/flaskr/service/shifu/struct_uils.py

```
@@ -0,0 +1,19 @@
+
+
+def find_node_with_parents(
+    root: HistoryItem, target_bid: str, current_path: Optional[list[HistoryItem]] = None
+) -> Optional[list[HistoryItem]]:
+    if current_path is None:
+        current_path = []
+    current_path.append(root)
+    if root.bid == target_bid:
+        return current_path.copy()
+
+    for i, child in enumerate(root.children):
```

Ruff (0.12.2)

Loop control variable `i` not used within loop body

Rename unused `i` to `_i`

(B007)